python/cudf/cudf/_lib/pylibcudf/lists.pyx

```diff
@@ -33,3 +41,55 @@ cpdef Table explode_outer(Table input, size_type explode_column_idx):
         c_result = move(cpp_explode.explode_outer(input.view(), explode_column_idx))
 
     return Table.from_libcudf(move(c_result))
+
+
+cpdef Column concatenate_rows(Table input):
+    """Row-wise concatenating multiple lists columns into a single lists column.
+
+    Parameters
+    ----------
+    input : Table
+        The input table
+
+    Returns
+    -------
+    Table
+        A new Column of concatenated rows
+    """
+    cdef unique_ptr[column] c_result
+
+    with nogil:
+        c_result = move(cpp_concatenate_rows(input.view()))
+
+    return Column.from_libcudf(move(c_result))
+
+
+cpdef Column concatenate_list_elements(Column input, bool dropna):
+    """Concatenating multiple lists on the same row into a single list.
+
+    Parameters
+    ----------
+    input : Column
+        The input column
+
+    Returns
+    -------
+    Column
+        A new Column of concatenated list elements
+    dropna : bool
+        If true, null list elements will be ignored
+        from concatenation. Otherwise the row will be nulled out.
```

Comment:
```suggestion
        from concatenation. Otherwise any input null values will result in
        the corresponding output row being set to null.
```